### ISSUE TITLE
Search page periodicaly refreshes if long query entered

### ISSUE THREAD

**Issue by hc4** [Graylog2/graylog2-server#3611]
I have entered long query (~1500 url encoded chars)
Graylog displayed correct searh result, but start to refresh page every 5 seconds.
In browser console there is logged same error pattern:
![image](https://cloud.githubusercontent.com/assets/17475270/23912539/01ada190-08f1-11e7-9075-d62157f4fe18.png)

## Your Environment

* Graylog Version: 2.2.2
* Browser version: Yandex 16.11.0.2680

**Comment by edmundoa** [comment]
Hi @hc4,

Thank you for reporting this issue.

Would you be so kind as to add more information about this issue? I'm specially interested in the request that is receiving the 400 error as response (you can probably see which one it is in the network tab of your browser's developer tools), and any errors that may appear in the server logs while this situation occurs.

By the way, could you also try reproducing the issue in one of the [supported browsers](http://docs.graylog.org/en/2.2/pages/configuration/web_interface.html#browser-compatibility)? We don't support the Yandex browser.

**Comment by hc4** [comment]
Just found that is was caused by SQUID, which deny requests with URI >4KB (value hardcoded)
Issue could be closed.

**Comment by joschi** [comment]
@hc4 Thanks for the feedback!

**Comment by hc4** [comment]
Investigate furthermore and found that it seems the problem of graylog too.
Could you please reopen issue.

There is actual request, sent to greylog:
```
GET http://graylog/web/streams/5***8/search?rangetype=absolute&fields...<<total url length 1467 bytes>> HTTP/1.1
Authorization: Negotiate <<2258 bytes of negotiate request>>
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.100 YaBrowser/16.11.0.2680 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Accept-Encoding: gzip, deflate, sdch
Accept-Language: ru,en;q=0.8
Host: graylog
X-Forwarded-For: local-ip
Cache-Control: max-age=0
Connection: keep-alive
Remote-User: user
X-Graylog-Server-URL: http://graylog
```

Total request length 4235 bytes

And graylog-server response:
```
2017/03/27 18:48:09.816| 11,2| http.cc(735) processReplyHeader: HTTP Server local=127.0.0.1:34048 remote=127.0.0.1:80 FD 20 flags=1
2017/03/27 18:48:09.816| 11,2| http.cc(736) processReplyHeader: HTTP Server REPLY:
---------
HTTP/1.1 400 Bad Request
Date: Mon, 27 Mar 2017 15:48:09 GMT
Connection: close
Content-Length: 0


----------
```

The same problem on IE, Firefox and Chrome.
No errors in graylog logs. Tried to set debug level for graylog - no valueable information. Sockets logging level doesn't affect anything.

**Comment by kroepke** [comment]
To properly fix this we need to rewrite how searches are done. The refresh likely comes from the repeating searches exceeding some URL limit by adding a parameter or so.
Currently I don't see an easy fix for 2.2.3, so I'll move it out of the milestone.

**Comment by hc4** [comment]
Why not just increase limits for max request and url length?
Chromesupports URL's up to 32K

**Comment by joschi** [comment]
@hc4 You can increase that yourself for Grizzly (the web server used in Graylog):

* https://github.com/Graylog2/graylog2-server/blob/2.2.2/misc/graylog.conf#L118-L119
* https://grizzly.java.net/docs/2.3/apidocs/org/glassfish/grizzly/http/server/NetworkListener.html#setMaxHttpHeaderSize-int-

**Comment by joschi** [comment]
@hc4 Out of curiosity, how exactly is your query composed? Would it help if Graylog supported CIDR queries or if you were able to use some sort of lookup table?

**Comment by hc4** [comment]
Currently we combine a lot of substrings, which must be present in logs to gather some overview of known errors.
Thanks for info about increasing limits. Hope it'll solve problem.

**Comment by hc4** [comment]
My problem is solved.
But it seems, that graylog still has bug, which causes page reload in case of long request.
Ideally there is should be some error page displayed in such situations.

**Comment by dennisoelkers** [comment]
This has been fixed in `3.1.0` by OSSing the views, which are now the default search in `3.2.0`.
